stablehlo/dialect/TypeInference.cpp

```diff
@@ -3857,7 +3849,19 @@ LogicalResult verifyScatterOp(std::optional<Location> location,
   });
   bool scatterIndicesTypeRanked = scatterIndicesType.isa<RankedTensorType>();
 
-  // P1.
+  // scatter_c1
```

Comment:
Also, `scatter_c6`

Reply:
```
(C6) `element_type(updates[k]) = element_type(inputs[k])` for all k $\in$ [0, N).
```
This check alone cannot verify `scatter_c6`, but is verified later in `verifyReducerShape` which I've correctly labeled now.

Reply:
makes sense.

Reply:
makes sense.